src/aws-redshiftserverless/namespace.ts

```diff
@@ -0,0 +1,356 @@
+    this.iamRoles = props.iamRoles ?? [];
+
+    this.validateAdmin();
+    this.validateDbName();
+    this.validateFinalSnapshot();
+    this.validateDefaultIamRole();
+    this.validateNamespaceName();
+
+    const namespace = this.createNamespace(this, 'Resource', {
+      adminUsername: this.props.adminUsername,
+      adminUserPassword: this.props.adminUserPassword?.unsafeUnwrap(),
+      dbName: this.props.dbName,
+      defaultIamRoleArn: this.props.defaultIamRole?.roleArn,
+      finalSnapshotName: this.props.finalSnapshotName,
+      finalSnapshotRetentionPeriod: this.props.finalSnapshotRetentionPeriod,
+      iamRoles: Lazy.list({ produce: () => this.iamRoles.map(role => role.roleArn) }, { omitEmpty: true }),
+      kmsKeyId: this.props.kmsKey?.keyId,
+      logExports: this.props.logExports,
+      namespaceName: this.physicalName,
+    });
+
+    this.namespaceArn = namespace.attrNamespaceNamespaceArn;
+    this.namespaceName = namespace.attrNamespaceNamespaceName;
+    this.namespaceId = namespace.attrNamespaceNamespaceId;
+  }
+
+  protected createNamespace(
+    scope: Construct,
+    id: string,
+    props: aws_redshiftserverless.CfnNamespaceProps,
+  ): aws_redshiftserverless.CfnNamespace {
+    return new aws_redshiftserverless.CfnNamespace(scope, id, props);
+  }
+
+  /**
+   * Validates admin settings.
+   */
+  private validateAdmin(): void {
+    const adminUsername = this.props.adminUsername;
+    const adminUserPassword = this.props.adminUserPassword;
+
+    if (Token.isUnresolved(adminUsername)) {
+      return;
+    }
+
+    if (
+      (adminUsername !== undefined && adminUserPassword === undefined) ||
+      (adminUsername === undefined && adminUserPassword !== undefined)
+    ) {
+      throw new Error('You must specify both `adminUsername` and `adminUserPassword`, or neither.');
+    }
+
+    if (adminUsername && !/^[a-zA-Z][a-zA-Z_0-9+.@-]*$/.test(adminUsername)) {
+      throw new Error(
+        `\`adminUsername\` must start with a letter and can only contain letters, numbers, and the special characters: _, +, ., @, -, got: ${adminUsername}.`,
+      );
+    }
+  }
+
+  /**
+   * Validates a database name.
+   */
+  private validateDbName(): void {
+    const dbName = this.props.dbName;
+
+    if (Token.isUnresolved(dbName) || dbName === undefined) {
+      return;
+    }
+
+    if (!/^[a-zA-Z][a-zA-Z_0-9+.@-]*$/.test(dbName) || dbName.length > 127) {
+      throw new Error(
+        `\`dbName\` must start with a letter, can only contain letters, numbers, and the special characters: _, +, ., @, -, and must not exceed 127 characters, got: ${dbName}.`,
+      );
+    }
+  }
+
+  /**
+   * Validates final snapshot settings.
+   */
+  private validateFinalSnapshot(): void {
+    const finalSnapshotName = this.props.finalSnapshotName;
+    const finalSnapshotRetentionPeriod = this.props.finalSnapshotRetentionPeriod;
+
+    if (Token.isUnresolved(finalSnapshotName)) return;
+
+    if (finalSnapshotName) {
```

Comment:
For the case of empty string.

```suggestion
    if (finalSnapshotName !== undefined) {
```